File: frontend/calendar_app/lib/models/employee.dart
class Employee {
  final String id;
  final String firstName;
  final String lastName;
  final String email;
  final String jobTitle;
  final String companyId;
  final String companyName;
  final String mobilePhone; // New property

  Employee({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.jobTitle,
    required this.companyId,
    this.companyName = '',
    this.mobilePhone = '',
  });

  String get fullName => '$firstName $lastName';

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'],
      firstName: json['firstName'],
      lastName: json['lastName'],
      email: json['email'],
      jobTitle: json['jobTitle'],
      companyId: json['companyId'],
      companyName: json['companyName'] ?? '',
      mobilePhone: json['mobilePhone'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'jobTitle': jobTitle,
      'companyId': companyId,
      'companyName': companyName,
      'mobilePhone': mobilePhone,
    };
  }
}

File: backend/Domain/Entities/Employee.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities
{
    public class Employee : User
    {
        public Guid CompanyId { get; private set; }
        public virtual Company Company { get; private set; }
        public string JobTitle { get; private set; }
<<<<<<< Updated upstream
        public string MobilePhone { get; private set; } 
=======
        public string MobilePhone { get; private set; } // Add this property
>>>>>>> Stashed changes
        public virtual ICollection<CalendarEvent> Events { get; private set; }
        
        // For EF Core
        private Employee() : base() { }
        
        public Employee(string firstName, string lastName, string email, string passwordHash, 
                      Guid companyId, string jobTitle, string mobilePhone = "") 
            : base(firstName, lastName, email, passwordHash)
        {
            CompanyId = companyId;
            JobTitle = jobTitle ?? throw new ArgumentNullException(nameof(jobTitle));
            MobilePhone = mobilePhone ?? "";
            Events = new List<CalendarEvent>();
        }
        
        public void UpdateJobTitle(string jobTitle)
        {
            if (string.IsNullOrWhiteSpace(jobTitle))
                throw new ArgumentException("Job title cannot be empty", nameof(jobTitle));
                
            JobTitle = jobTitle;
        }

        public void UpdateMobilePhone(string mobilePhone)
        {
            MobilePhone = mobilePhone ?? "";
        }
    }
}
